### PR TITLE
Optimized names to save gas

### PR DESCRIPTION
**Pull request by filmakarov**
Optimization for Method names.

Basic info: in smart contracts, there is a difference in the order of the functions. The later the sort will consume more. Each position will have an extra 22 gas.

Details: https://medium.com/joyso/solidity-how-does-function-name-affect-gas-consumption-in-smart-contract-47d270d8ac92
Tool: https://emn178.github.io/solidity-optimize-name/ 

Made optimization for execTransaction, executeCall, executeBatchCall, validateUserOp (most used functions) so their method names start with zeros.

### CODE REVIEW

File: contracts/smart-contract-wallet/BaseSmartAccount.sol [hunk 1]
```diff
@@ -55,12 +55,13 @@ abstract contract BaseSmartAccount is IAccount, BaseSmartAccountErrors {
...
      * subclass doesn't need to override this method. Instead, it should override the specific internal validation methods.
+     * @dev name is optimized for this method to be cheaper to be called
      */
-    function validateUserOp(
+    function validateUserOp_kab(
         UserOperation calldata userOp,
         bytes32 userOpHash,
         uint256 missingAccountFunds
-    ) external virtual override returns (uint256 validationData) {
+    ) public returns (uint256 validationData) {
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/BaseSmartAccount.sol:64]: Entry point will always call validateUserOp on the account as per IAccount interface, so we should / need not keep this one.

File: contracts/smart-contract-wallet/SmartAccount.sol [hunk 2]
```diff
@@ -271,16 +271,17 @@ contract SmartAccount is
...
      * @param refundInfo Required information for gas refunds
      * @param signatures Packed signature/s data ({bytes32 r}{bytes32 s}{uint8 v})
      */
-    function execTransaction(
+    function execTransaction_S6W(
         Transaction memory _tx,
         FeeRefund memory refundInfo,
         bytes memory signatures
-    ) external payable virtual override returns (bool success) {
+    ) public payable returns (bool success) {
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/SmartAccount.sol:284]: can we keep it virtual, that way we can override this as well.

File: contracts/smart-contract-wallet/SmartAccount.sol [hunk 3]
```diff
@@ -641,25 +650,38 @@ contract SmartAccount is
     }
 
     /**
-     * execute a transaction (called directly from owner, or by entryPoint)
+     * Execute a transaction (called directly from owner, or by entryPoint)
+     * @dev name is optimized for this method to be cheaper to be called
      */
-    function executeCall(
+    function executeCall_s1m(
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/SmartAccount.sol:656]: do add the test case for calling this method. in aa transactions calldata is prepared for executeCall and executeBatchCall, replace these instead.

Comment (livingrockrises) [reply contracts/smart-contract-wallet/SmartAccount.sol:656]: can we also manage to get some gas usage numbers for the changes?

File: contracts/smart-contract-wallet/test/upgrades/SmartAccount10.sol [hunk 4]
```diff
@@ -222,11 +222,11 @@ contract SmartAccount10 is
     /// @param _tx Wallet transaction
     /// @param refundInfo Required information for gas refunds
     /// @param signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v})
-    function execTransaction(
+    function execTransaction_S6W(
         Transaction memory _tx,
         FeeRefund memory refundInfo,
         bytes memory signatures
-    ) public payable virtual override returns (bool success) {
+    ) public payable returns (bool success) {
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/test/upgrades/SmartAccount10.sol:229]: virtual

File: contracts/smart-contract-wallet/test/upgrades/SmartAccount11.sol [hunk 5]
```diff
@@ -277,11 +277,11 @@ contract SmartAccount11 is
      * @param refundInfo Required information for gas refunds
      * @param signatures Packed signature/s data ({bytes32 r}{bytes32 s}{uint8 v})
      */
-    function execTransaction(
+    function execTransaction_S6W(
         Transaction memory _tx,
         FeeRefund memory refundInfo,
         bytes memory signatures
-    ) external payable virtual override returns (bool success) {
+    ) public payable returns (bool success) {
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/test/upgrades/SmartAccount11.sol:284]: virtual

File: contracts/smart-contract-wallet/test/upgrades/SmartAccount4.sol [hunk 6]
```diff
@@ -221,11 +221,11 @@ contract SmartAccount4 is
     /// @param _tx Wallet transaction
     /// @param refundInfo Required information for gas refunds
     /// @param signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v})
-    function execTransaction(
+    function execTransaction_S6W(
         Transaction memory _tx,
         FeeRefund memory refundInfo,
         bytes memory signatures
-    ) external payable virtual override returns (bool success) {
+    ) public payable returns (bool success) {
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/test/upgrades/SmartAccount4.sol:228]: virtual

File: contracts/smart-contract-wallet/test/upgrades/SmartAccount7.sol [hunk 7]
```diff
@@ -231,11 +231,11 @@ contract SmartAccount7 is
     /// @param _tx Wallet transaction
     /// @param refundInfo Required information for gas refunds
     /// @param signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v})
-    function execTransaction(
+    function execTransaction_S6W(
         Transaction memory _tx,
         FeeRefund memory refundInfo,
         bytes memory signatures
-    ) external payable virtual override returns (bool success) {
+    ) public payable returns (bool success) {
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/test/upgrades/SmartAccount7.sol:238]: virtual

File: contracts/smart-contract-wallet/test/upgrades/SmartAccount9.sol [hunk 8]
```diff
@@ -222,11 +222,11 @@ contract SmartAccount9 is
     /// @param _tx Wallet transaction
     /// @param refundInfo Required information for gas refunds
     /// @param signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v})
-    function execTransaction(
+    function execTransaction_S6W(
         Transaction memory _tx,
         FeeRefund memory refundInfo,
         bytes memory signatures
-    ) public payable virtual override returns (bool success) {
+    ) public payable returns (bool success) {
```
Comment (livingrockrises) [inline contracts/smart-contract-wallet/test/upgrades/SmartAccount9.sol:229]: virtual

File: test/smart-wallet/testGroup1.ts [hunk 9]
```diff
@@ -339,7 +339,7 @@ describe("Base Wallet Functionality", function () {
     await expect(
       userSCW
         .connect(accounts[0])
-        .execTransaction(transaction, refundInfo, signature)
+        .execTransaction_S6W(transaction, refundInfo, signature)
```
Comment (livingrockrises) [inline test/smart-wallet/testGroup1.ts:342]: just at one place try to get gas usage improvements. by calling execTransaction, execTransaction_S6W both for two different similar transactions.

File: contracts/smart-contract-wallet/BaseSmartAccount.sol (hunk 1 above)
Comment (filmakarov) [reply contracts/smart-contract-wallet/BaseSmartAccount.sol:64]: you mean there's no way anything will call `validateUserOp_kab` directly?

File: contracts/smart-contract-wallet/SmartAccount.sol (hunk 2 above)
Comment (filmakarov) [reply contracts/smart-contract-wallet/SmartAccount.sol:284]: +

File: contracts/smart-contract-wallet/SmartAccount.sol (hunk 3 above)
Comment (filmakarov) [reply contracts/smart-contract-wallet/SmartAccount.sol:656]: +

executeCall · 39107 · 169358 · **65077**
executeCall_s1m · 39085 · 169336 · **65055**
**22 gas less on average**

executeBatchCall · 52026 · 75260 · **63739**
executeBatchCall_4by · 51972 · 75206 · **63685**
**54 gas less on average**

File: test/smart-wallet/testGroup1.ts (hunk 9 above)
Comment (filmakarov) [reply test/smart-wallet/testGroup1.ts:342]: done
Screen Shot 2023-03-14 at 13 36 01

File: contracts/smart-contract-wallet/BaseSmartAccount.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/smart-contract-wallet/BaseSmartAccount.sol:64]: yes, its very unlikely that any entity except entry point will call this method. and entry point will always call validateUserOp

Comment (filmakarov) [reply contracts/smart-contract-wallet/BaseSmartAccount.sol:64]: removed

**Comment by filmakarov** [comment]
I re-requested review, are you able to merge @livingrockrises ?

**Review by livingrockrises — APPROVED** [review]
LGTM

**Comment by livingrockrises** [comment]
merged

**Pull request merged**